base/dcp_sharded.go

```diff
@@ -0,0 +1,337 @@
+	} else {
+		return false, "", nil
+	}
+}
+
+// createCBGTManager creates a new manager for a given bucket and bucketSpec
+// Inline comments below provide additional detail on how cbgt uses each manager
+// parameter, and the implications for SG
+func initCBGTManager(dbName string, bucket Bucket, spec BucketSpec) (*CbgtContext, error) {
+
+	// uuid: Unique identifier for the node. Used to identify the node in the config
+	// TODO: Without UUID persistence across SG restarts, a restarted SG node relies on heartbeater to remove
+	// 		 the previous version of that node from the cfg, and assign pindexes to the new one.
+	// 		(note that in a single node scenario, this is latency removing previous version of itself
+	//  	on restart, if time between restarts is less than heartbeat expiry time).
+	uuid := cbgt.NewUUID()
+
+	cfgCB, err := initCfgCB(bucket, spec)
+	if err != nil {
+		Warnf(KeyDCP, "Error initializing cfg for import sharding: %v", err)
+		return nil, err
+	}
+
+	// tags: Every node participating in sharded DCP has feed, janitor, pindex and planner roles
+	//   	"feed": runs dcp feed
+	//   	"janitor": maintains feed instances, attempts to match to plan
+	//   	"pindex": handles DCP events
+	//   	"planner": assigns partitions to nodes
+	// TODO: Every participating node acts as planner, which means every node is going to be trying to
+	//       reassign partitions (vbuckets) to nodes when a cfg change is detected, and relying on cas
+	//       failure in the cfg to avoid rework.  cbft doesn't do this -
+	//       it leverages ns-server to identify a master node.
+	//       Could revisit in future, as leader-elect style functionality is
+	//       required for sg-replicate HA anyway
+	tags := []string{"feed", "janitor", "pindex", "planner"}
+
+	// weight: Allows for weighted distribution of vbuckets across participating nodes.  Not
+	//         used by Sync Gateway
+	weight := 1
+
+	// container: Used by cbgt to determine node hierarchy.  Not needed by Sync Gateway
+	container := ""
+
+	// extras: Can be used to pass Sync Gateway specific node information to callbacks.  Not
+	//         currently needed by Sync Gateway.
+	extras := ""
+
+	// bindHttp: Used for REST binding (not needed by Sync Gateway), but also as a unique identifier
+	//           in some circumstances.  See below for additional information.
+	// 		(from accel): use the uuid as the bindHttp so that we don't have to make the user
+	// 		configure this, since we're just using for uniqueness and not for REST API
+	// 		More info here:
+	//   		https://github.com/couchbaselabs/cbgt/issues/1
+	//   		https://github.com/couchbaselabs/cbgt/issues/25
+	bindHttp := uuid
+
+	// serverURL: Passing gocb connect string.
+	serverURL, err := spec.GetGoCBConnString()
+	if err != nil {
+		return nil, err
+	}
+
+	// dataDir: file system location for files persisted by cbgt.  Needs to be unique
+	// per database
+
+	dataDir := os.TempDir() + "sg_" + dbName
```

Comment:
I think we want a directory separator here? `os.TempDir()` returns `"/tmp"` on linux, with no trailing slash.

```suggestion
	dataDir := filepath.Join(os.TempDir(), "sg_" + dbName)
```